dangerzone/cli.py

```diff
@@ -108,7 +108,7 @@ def cli_main(
     if documents_failed != []:
         print_header("Failed to convert document(s)")
         for document in documents_failed:
-            click.echo(document.input_filename)
+            click.echo(click.format_filename(document.input_filename))
```

Comment:
> I'd like us to have a sanitation function that does the above

Sorry, please correct if I misunderstood - I should be using `utils.replace_control_chars()` instead of `click.format_filename()`? Or there should be a separate sanitation function added that does what `click.format_filename()` does but is not related to Click?

Reply:
Whoops, sorry, missed your comment. You're right, I was not clear because I don't have a clear solution to this problem yet. Here's what I think:

1. We have this sanitation need both in CLI code, and in library code. So we definitely need a Click-agnostic function, for consistency.
2. We already have `utils.replace_control_chars()`, but it replaces every non-printable character with "_". This is a bad solution for non-english filenames, but we tolerate it for now, for security reasons.
3. `click.format_filename()` deals with invalid bytes, but does not replace control characters. You can check the following in a Python interpreter, and see that your terminal with interpret the ANSI codes:

      ```python
      print(click.format_filename("\033[31;1;4m BaD TeΧt \u200d ❌ \033[0m"))
      ```

So, the ideal solution would be to have a non-Click function that can:
* Replace control characters AND invalid bytes with a placeholder (� is nice).
* Retain valid Unicode letters that do not belong to the English alphabet.

If you have some time, I would suggest looking into [`unicodedata.category`](https://docs.python.org/3/library/unicodedata.html#unicodedata.category), and the associated [General Category values](https://www.unicode.org/reports/tr44/#General_Category_Values). I think that if we update `util.replace_control_chars()` to replace every Unicode character that belongs to the `C*` / `Zl` / `Zp` categories with �, we would have a pretty good solution.

Reply:
I've updated `replace_control_chars` to use General Category values in 3212846c82de60083b4c75a810ac1f28f2238110.

Reply:
Yeap, I think it's awesome, thanks for the change!